macSKK/KeyBindingSet.swift

```diff
@@ -4,19 +4,51 @@
 import AppKit
 
 struct KeyBindingSet {
-    let values: [KeyBinding.Action: [KeyBinding.Input]]
-    let dict: [KeyBinding.Input: KeyBinding.Action]
+    /**
+     * 修飾キーを除いたキー入力が同じ場合は修飾キーが多いものが前に来るように並べた配列。
+     * 入力に一番合致するキー入力を返すために最初にソートしてもっておく。
+     */
+    let sorted: [(KeyBinding.Input, KeyBinding.Action)]
 
     static let defaultKeyBindingSet = KeyBindingSet(KeyBinding.defaultKeyBindingSettings)
 
     init(_ values: [KeyBinding]) {
-        self.values = Dictionary(uniqueKeysWithValues: values.map { ($0.action, $0.inputs) })
-        self.dict = Dictionary(uniqueKeysWithValues: values.flatMap { keyValue in
+        sorted = values.flatMap { keyValue in
             keyValue.inputs.map { ($0, keyValue.action) }
+        }.sorted(by: { lts, rts in
+            if lts.0.key == rts.0.key {
+                return lts.0.modifierFlags.rawValue > rts.0.modifierFlags.rawValue
+            } else {
+                switch (lts.0.key, rts.0.key) {
+                case let (.character(l), .character(r)):
+                    return l < r
+                case let (.code(l), .code(r)):
+                    return l < r
+                case (.character, .code):
+                    return false
+                case (.code, .character):
+                    return true
+                }
+            }
         })
     }
 
     func action(event: NSEvent) -> KeyBinding.Action? {
-        return dict[KeyBinding.Input(event: event)]
+        let input = KeyBinding.Input(event: event)
+        for keyBind in sorted {
+            if keyBind.0.key == input.key {
+                switch input.key {
+                case .character(let c):
+                    if KeyBinding.Key.characters.contains(c) && keyBind.0.modifierFlags == input.modifierFlags {
```

Comment:
`KeyBinding.Key.characters.contains(c)`はこれ`input.key`のコンストラクト時 👇 にもチェックしています。

https://github.com/mtgto/macSKK/blob/18db2f0fe429ccebc37504b98037a62826ab9f99/macSKK/KeyBinding.swift#L107-L109

よって`.character`のケースに進入した以上はもうこのチェックは不要ということで

```suggestion
                    keyBind.0.modifierFlags == input.modifierFlags {
```

☝️ こうしても大丈夫なような 🤔